src/Shared/ServerInfrastructure/ManualResetValueTaskSource.cs

```diff
@@ -21,5 +21,15 @@ internal sealed class ManualResetValueTaskSource<T> : IValueTaskSource<T>, IValu
         void IValueTaskSource.GetResult(short token) => _core.GetResult(token);
         public ValueTaskSourceStatus GetStatus(short token) => _core.GetStatus(token);
         public void OnCompleted(Action<object> continuation, object state, short token, ValueTaskSourceOnCompletedFlags flags) => _core.OnCompleted(continuation, state, token, flags);
+
+        public ValueTaskSourceStatus GetStatus() => _core.GetStatus(_core.Version);
+
+        public void TrySetResult(T result)
```

Comment:
Can we just remove this and call SetResult everywhere?

Reply:
It is required. If a queue connection awaitable was completed in abort we don't want to try to complete it again when it is dequeued. If you really want I can get rid of this method and move the status test there, but the status test needs to happen somewhere.